### PR TITLE
Feature/3521 idea hub remove local setting

### PR DESCRIPTION
**Pull request by ivankruchkoff**
## Summary

Addresses issue #3521 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/idea-hub/components/settings/SettingsView.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		dashboardPermalink
+	);
+
+	return (
+		<p
+			dangerouslySetInnerHTML={ sanitizeHTML( content, {
+				ALLOWED_TAGS: [ 'a' ],
+				ALLOWED_ATTR: [ 'href' ],
+			} ) }
+		/>
```
Comment (eugene-manuilov) [inline assets/js/modules/idea-hub/components/settings/SettingsView.js:47]: We need to use `createInterpolateElement`: https://github.com/google/site-kit-wp/blob/80868c0780a1d9febc153adadea50617624a22db/assets/js/components/OptIn.js#L77-L88

File: includes/Modules/Idea_Hub.php [hunk 2]
```diff
@@ -40,7 +38,7 @@
  * @ignore
  */
 final class Idea_Hub extends Module
-	implements Module_With_Scopes, Module_With_Settings, Module_With_Debug_Fields, Module_With_Assets {
+	implements Module_With_Scopes, Module_With_Assets {
```
Comment (eugene-manuilov) [inline includes/Modules/Idea_Hub.php:41]: We would also need to delete the `Module_With_Settings_Trait` trait usage and `on_deactivation` and `setup_settings` methods if we removed the `Module_With_Settings` interface.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @ivankruchkoff. Almost looks good to me. Just a few things to address before we can merge it.

File: assets/js/modules/idea-hub/components/settings/SettingsView.js (hunk 1 above)
Comment (ivankruchkoff) [reply assets/js/modules/idea-hub/components/settings/SettingsView.js]: Updated in https://github.com/google/site-kit-wp/pull/3557/commits/e0b9a129ce24f2b3146d3d5f4e83fcd7385f0a4c

File: includes/Modules/Idea_Hub.php (hunk 2 above)
Comment (ivankruchkoff) [reply includes/Modules/Idea_Hub.php:37]: Removed in https://github.com/google/site-kit-wp/pull/3557/commits/dec1ebae3fc526df0df1491e17832d2567e3a098

File: assets/js/modules/idea-hub/components/settings/SettingsView.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from '../../datastore/constants';
+import Link from '../../../../components/Link';
+import { createInterpolateElement } from '@wordpress/element';
```
Comment (eugene-manuilov) [inline assets/js/modules/idea-hub/components/settings/SettingsView.js:30]: This should go within the WordPress dependencies section.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Looks good to me. There is the last thing we need to fix and it should be good to go. Thanks, @ivankruchkoff.

File: assets/js/modules/idea-hub/components/settings/SettingsView.js (hunk 3 above)
Comment (ivankruchkoff) [reply assets/js/modules/idea-hub/components/settings/SettingsView.js]: https://github.com/google/site-kit-wp/pull/3557/commits/62aa044ad7691b28a13d3f21c7eb95293a245b44

**Review by eugene-manuilov — APPROVED** [review]

**Pull request merged**